### PR TITLE
New PM and help popups in Curve

### PR DESCRIPTION
**Pull request by marcusforsberg**
I'm sure the JS can be prettified by a more skilled JS developer than I, but I've got a lot of positive feedback on how these work and look, so here goes! :)

Preview

### CODE REVIEW

File: Themes/default/scripts/script.js [hunk 1]
```diff
@@ -317,22 +317,63 @@ String.prototype.easyReplace = function (oReplacements)
 	return sResult;
 }
 
-
-// Open a new window.
+// Open a new window (only exists for backwards compatibility) - DEPRECATED
```
Comment (live627) [inline Themes/default/scripts/script.js:320]: Can I ask why you decided to deprecate this?

**Comment by marcusforsberg** [comment]
Because it'd be a much better idea to replace all calls to reqWin() to ones that don't call a URL but instead just output the text string immediately, but I'm not sure we actually wanna do that and besides, there are still mods that use reqWin() who need to be supported - But it shouldn't really be used the way it is.

**Comment by Spuds** [comment]
Looks very nice ... a couple of passing questions while I have better than a 1X connection !

1) I don't think the ajax_indicator(true/false); bar is useful for this, either the "popup" appears or it does not, no need for the extra screen info as it just detracts from the experience.  If you really want you could pop the div and put the spinner in that and then replace it with the content when it arrives.

2) It would be good to get someone who is knowledgeable with ally etc if this change causes any problem (well worse than we had that is).  I'm thinking since its just an overlay div it will be fine, but I don't really know.

3) General usability question ... should we have a close button or a title on the X icon for the hover info?  Granted its not a popup, nor modal, and it closes with esc or a browser window mouse click.  Just asking to be sure its OK for accessibility, it may be just fine ;)

**Comment by marcusforsberg** [comment]
Thanks!

1) I partially agree here but the thing is there is technically an AJAX call going on in there. I didn't have it in there at first but then I tried it on a live site and I noticed it took a little while for the popup to load since it had to load the external URL and all that first. Therefore it made sense to me to put in that loading bar in there so the user knows that something IS happening, even though it might take a second. But you do have a valid point, it may draw your attention to the top too much!

2) Not sure what ally refers to here? :o I did notice a stupid bug/oversight on the registration page... If you want to listen to the captcha it uses reqWin() for that but since it has that overlay on it you can't type in the letters as you listen. And the "play again" button is broken. I'm of course going to roll out a fix for that ASAP. :)

3) One might argue that a description for the X would be in place but I don't really thing it's needed. If anything it'd make it more cluttered, in my eyes. A title attribute probably wouldn't hurt at all though. But since it closes on both click and ESC I think most users will figure it out quite easily.

Thanks a lot for the feedback! :)

**Comment by Spuds** [comment]
1) .. yeah the spinner or throbber is a good feedback to show the user "I'm working on your request" ... so adding that to the window you created instead of SMF's rather clumsy ajax bar, might be a nice improvement (hint hint, nudge nudge :P )

2) Thanks for the catching that bug and the update :) ... ally is lots of things, but in this context it was screen reading for the blind, dyslexic or other visually challenged.  SMF is not great for those with disabilities (in general) but we don't want to go the wrong way unnecessarily,  Ant would know one way or another on this if you can catch him on IM

**Comment by marcusforsberg** [comment]
1) I've thrown a few angry eyes on the ajax notification before, I might try to fix that someday. But I do agree, a loading indicator in the window itself is probably a better idea.

2) Oh I see. My knowledge on the topic doesn't suggest anything other than a title tag for the close button, I'd say that'd be enough in that case, since, as you said, it's just a div overlay. Other than fixing the bug, of course.

**Comment by ghost** [comment]
Why the dark overlay? If its just a info popup, it should not be necessary to darken everything else? I feel that should be reserved for _required_ interaction - so that everything else really is _on hold_.

Stopping me from continung just to read that tid-bit of info would really annoy me lol.

**Comment by marcusforsberg** [comment]
I partially agree and originally didn't even put that overlay in for help popups, only for PMs, but I showed them to a bunch of people and they all seemed to prefer the overlay to be there all the time to put focus to the content. I'm fine with either alternative, though!

**Comment by live627** [comment]
TBH, I would prefer the overlay there regardless. But I'm not managing this project, so....

**Comment by marcusforsberg** [comment]
I'm starting to realise how broken this is, now. I must have been very tired while creating it. :P I was, for some reason, under the impression that reqWin() was only used for the help popups. I forgot about the error log and captcha and install.php. Will work on a fix for it...

**Pull request merged**